crates/cargo-bsp/src/cargo_communication/request_actor_unit_graph.rs

```diff
@@ -0,0 +1,109 @@
+use bsp_types::requests::Request;
+use bsp_types::StatusCode;
+use log::warn;
+use serde::Deserialize;
+
+use crate::cargo_communication::cargo_types::cargo_command::CreateCommand;
+use crate::cargo_communication::cargo_types::cargo_result::CargoResult;
+use crate::cargo_communication::cargo_types::event::{CargoMessage, Event};
+use crate::cargo_communication::cargo_types::unit_graph::UnitGraph;
+use crate::cargo_communication::request_actor::{CargoHandler, RequestActor};
+
+#[derive(PartialEq)]
+pub enum UnitGraphStatusCode {
+    Ok,
+    Cancelled,
+}
+
+impl<R, C> RequestActor<R, C>
+where
+    R: Request,
+    R::Params: CreateCommand,
+    R::Result: CargoResult,
+    C: CargoHandler<CargoMessage>,
+{
+    pub fn run_unit_graph(&mut self) -> UnitGraphStatusCode {
+        self.report_task_start(self.state.root_task_id.clone(), None, None);
+        self.report_task_start(
+            self.state.unit_graph_state.task_id.clone(),
+            Some("Started unit graph command".to_string()),
+            None,
+        );
+
+        let mut received_unit_graph = false;
+
+        while let Some(event) = self.next_event() {
+            match event {
+                Event::Cancel => {
+                    self.cancel();
+                    return UnitGraphStatusCode::Cancelled;
+                }
+                Event::CargoFinish => {
+                    self.finish_unit_graph(received_unit_graph);
+                    break;
```

Comment:
chcemy zwracać `UnitGraphStatusCode::Ok` nawet jeśli nie otrzymaliśmy odpowiedniego info?

Reply:
Jeśli się jakimś cudem nie powiedzie unit graph, to chyba w żaden sposób nie stopuje normalnej komendy, po prostu nie będzie podawany progress kompilacji

Reply:
brzmi sensownie